lstchain/tools/lstchain_create_drs4_pedestal_file.py

```diff
@@ -0,0 +1,122 @@
+"""
+Create drs4 time pedestal fits file.
+"""
+import numpy as np
+from astropy.io import fits
+from tqdm.autonotebook import tqdm
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe.io import EventSource
+from lstchain.calib.camera.r0 import LSTR0Corrections
+from lstchain.calib.camera.drs4 import DragonPedestal
+
+
+class PedestalFITSWriter(Tool):
+
+    name = "PedestalFITSWriter"
+    description = "Generate a pedestal FITS file"
+
+    output = traits.Path(
+        default_value="pedestal.fits",
+        help="Path to the generated fits pedestal file",
+        directory_ok=False,
+        exists=False,
+    ).tag(config=True)
+
+    deltaT = traits.Bool(
+        help="Flag to use deltaT correction. Default=True", default_value=True
+    ).tag(config=True)
+
+    progress_bar = traits.Bool(
+        help="show progress bar during processing",
+        default_value=True,
+    ).tag(config=True)
+
+    aliases = {
+        "input": "EventSource.input_url",
+        "output": "PedestalFITSWriter.output_file",
+    }
+
+    flags = {
+        "deltaT": (
+            {"PedestalFITSWriter": {"deltaT": True}},
+            "Flag to use deltaT correction. Default is True",
+        )
+    }
+
+    classes = [EventSource, LSTR0Corrections, DragonPedestal]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a pedestal FITS file for low level calibration.
+
+        For getting help run:
+        lstchain_create_drs4_pedestal_file --help
+        """
+
+        self.eventsource = None
+        self.pixel_ids = None
+        self.pedestal = None
+        self.lst_r0 = None
+
+    def setup(self):
+
+        self.log.debug("Opening file")
+        self.eventsource = EventSource.from_config(parent=self)
+        self.lst_r0 = self.add_component(LSTR0Corrections(parent=self))
+
+    def start(self):
+
+        for event in self.eventsource:
+            tel_id = event.r0.tels_with_data[0]
+            self.pixel_ids = event.lst.tel[tel_id].svc.pixel_ids
+            self.pedestal = DragonPedestal(
+                tel_id=tel_id, n_module=event.lst.tel[tel_id].svc.num_modules
+            )
+            break
+
+        if self.deltaT:
+            self.log.info("DeltaT correction active")
+            self.log.info(f"Progress bar {self.progress_bar}")
+            for event in tqdm(
+                self.eventsource,
+                desc=self.eventsource.__class__.__name__,
+                total=self.eventsource.max_events,
+                unit="ev",
+                disable=not self.progress_bar,
+            ):
+                for tel_id in event.r0.tels_with_data:
+                    self.lst_r0.time_lapse_corr(event, tel_id)
```

Comment:
Is this line the only difference between `if self.deltaT` and `else`? If yes, use a single loop and move the if inside, much more readable.

Reply:
And one level more, then you get rid of the duplicated call to `fill_pedestal_event` and the `else` statement